### PR TITLE
fix(upload-link): Add better error message

### PR DESCRIPTION
**Pull request by Bento007**
- #2840

### Reviewers
**Functional:** @danieljhegeman 

**Readability:** @nayib-jose-gloria 

---

## Changes
- modify the error message returned when submitting a dataset while the dataset is still processing.

## QA steps (optional)
- Upload a dataset using a link. Then upload to the same dataset again using and upload-link. The response will be a 405 with the new error message.

### CODE REVIEW

**Review by nayib-jose-gloria — APPROVED** [review]

**Review by danieljhegeman — APPROVED** [review]
🚀

**Review by nayib-jose-gloria — APPROVED** [review]

**Review by danieljhegeman — COMMENTED** [review]
Shouldn't this tell them to _either_ cancel via DELETE endpoint _or_ wait until it has finished?

File: backend/corpora/lambdas/api/v1/collection_id/upload.py [hunk 1]
```diff
@@ -75,6 +75,10 @@ def upload_from_link(collection_id: str, token_info: dict, url: str, dataset_id:
     except NonExistentCollectionException:
         raise ForbiddenHTTPException()
     except InvalidProcessingStateException:
-        raise MethodNotAllowedException()
+        raise MethodNotAllowedException(
+            detail="Submission failed. A dataset cannot be updated while a previous update for the same dataset is in "
+            "progress. Please cancel the current submission with "
+            f"DELETE /v1/collections/collection/dataset/{dataset_id}"
```
Comment (nayib-jose-gloria) [inline backend/corpora/lambdas/api/v1/collection_id/upload.py:81]: actually, should we specify the mechanism for canceling? Curators have more user-friendly notebooks they could use instead if they'd like (to more easily set headers and whatnot), or the UI could be available. What do you think of just saying please cancel the current submission?

**Review by danieljhegeman — APPROVED** [review]
LGTM

**Pull request merged**